src/plugins/fs/ext.c

```diff
@@ -59,6 +59,75 @@ static guint32 fs_mode_util[BD_FS_MODE_LAST+1] = {
 
 #define UNUSED __attribute__((unused))
 
+static gint8 compute_percents(guint8 pass_cur, guint8 pass_total, gint val_cur, gint val_total)
+{
+    gint perc;
+    gint one_pass;
+    /* first get a percentage in the current pass/stage */
+    perc = (val_cur * 100) / val_total;
+
+    /* now map it to the total progress, splitting the stages equally */
+    one_pass = 100 / pass_total;
+    perc = ((pass_cur - 1) * one_pass) + (perc / pass_total);
+
+    return perc;
+}
+
+/**
+ * filter_line_fsck: (skip)
+ * Filter one line - decide what to do with it.
+ *
+ * Returns: Zero or positive number as a percentage, -1 if not a percentage, -2 on an error
+ */
+static gint8 filter_line_fsck(const gchar * line, guint8 total_stages, GError **error) {
+    static GRegex *regex;
+    static int reginit = 0;
+    GMatchInfo *match_info;
+    gint8 perc = -1;
+
+    if (!reginit) {
+        /* Compile regular expression that matches to e2fsck progress output */
+        regex = g_regex_new ("^([0-9][0-9]*) ([0-9][0-9]*) ([0-9][0-9]*) (/.*)", 0, 0, error);
+        if (regex == NULL) {
+            return -2;
+        }
+        reginit = 1;
+    }
+
+    /* Execute regular expression */
+    if (g_regex_match (regex, line, 0, &match_info)) {
+        guint8 stage;
+        long val_cur;
+        long val_total;
```

Comment:
``g_ascii_strtoll()`` returns a ``gint64``. And I think ``g_ascii_strtoull()`` would be better here, right?